### PR TITLE
Confluence-v1

### PR DESCRIPTION
**Pull request by DavidDwyer87**
This consists pull request consist of:
* confluence configuration.
* ascii documentation on the confluence configuration.
* readme describing how to configure the connector

### CODE REVIEW

File: confluence/confluence-v1.asciidoc [hunk 1]
```diff
@@ -0,0 +1,73 @@
+Confluence Rest Configuration
+----------------------------
+This documentation describs what data the confluence configuration retrives and the endpoints used.
```
Comment (chadanhuggup) [inline confluence/confluence-v1.asciidoc:3]: Small typo on describes, and retrieves

File: confluence/confluence-v1.asciidoc [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+This is this list of data that rest connector using confluence configuration will crawl.
+
+* spaces 
+* pages (such as Wiki pages), 
+* blog posts,  
+* pages and blog posts comments (indexed as metadata) 
+
+Authentication
+~~~~~~~~~~~~~~~
+Support Basic Authentication with username (atlasian account) and API Token
```
Comment (chadanhuggup) [inline confluence/confluence-v1.asciidoc:13]: small typo on Atlassian

File: confluence/confluence-v1.asciidoc [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+Supported Crawl
+~~~~~~~~~~~~~~~
+* Full Crawl supported
+* Re-Crawl supported - `rely on the strayContentDeletion`
+
+
+Rest Connector Variables used
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+* ${LW_BATCH_SIZE} - This variable is used to set the `limit` query parameter which controls the number results are return in the response for both `blogpost` and `pages`
```
Comment (chadanhuggup) [inline confluence/confluence-v1.asciidoc:28]: number **of** results **that** are returned

File: confluence/confluence-v1.asciidoc [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+* {LW_INDEX_START} - This variable  is used to set the `start` query parameter which use the travers the pagination. Confluence pagination is 0 based. This means the first page number is 0.
```
Comment (chadanhuggup) [inline confluence/confluence-v1.asciidoc:30]: You could add the $ at the beginning of the special variable.

Comment (chadanhuggup) [reply confluence/confluence-v1.asciidoc:30]: You could modify the description to the below:
"This variable  is used to set the `start` query parameter which is used to traverse the pagination. Confluence pagination is 0 index based. This means the first page number is 0"

File: confluence/confluence-v1.asciidoc [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+* {LW_PARAM_KEY} - This variable is uesd to access a value from the response from `main request`to used in a `additional request`. Confluence use case this variable added to URL to proform a GET request, to get comments for blogpost and pages
```
Comment (chadanhuggup) [inline confluence/confluence-v1.asciidoc:32]: You could add $ on the special variable

File: confluence/confluence-v1.asciidoc [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+*Notes:*
+
+List of objects (spaces, pages, blogs, comments) are retrieved in batches (25 by default). Configure query limit to get more objects (the max size is 250)
+
+
+[options="header"]
+|=======================
+|Endpoint|Http Verb |Query Param |Description |Request type
+|/wiki/rest/api/content|GET    |type=blogpost&start=0&limit=1&expand=body.storage|get all blogpost from confluence|Data Request
+|/wiki/rest/api/content/${LW_PARAM_KEY}/child/comment|GET|expand=body.storage|This endpoint gets all blogpost comment. The value of `id` from the main request needs to be assign to the `${LW_PARAM_KEY}` variable so that the addtional feature can build insert the value to the get url.  |Child Endpoint
```
Comment (chadanhuggup) [inline confluence/confluence-v1.asciidoc:48]: Small typo in "assigned"

Comment (chadanhuggup) [inline confluence/confluence-v1.asciidoc:48]: Could change "feature can build insert the value to the get url" to
"feature can insert the value when building the GET url"

File: confluence/confluence-v1.asciidoc [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+|/wiki/rest/api/content | GET |type=paget&start=0&limit=1&expand=body.storage |get all pages from confluence |Data Request
+|/wiki/rest/api/content/${LW_PARAM_KEY}/child/comment|GET|expand=body.storage|This endpoint gets all page comment. The value of `id` from the main request needs to be assign to the `${LW_PARAM_KEY}` variable so that the addtional feature can build insert the value to the get url.  |Child Endpoint
```
Comment (chadanhuggup) [inline confluence/confluence-v1.asciidoc:50]: Could change "comment" to "comments"
and 
"assign" to "assigned"

Could change the below
"addtional feature can build insert the value to the get url"
to
"additional feature can insert the value when building the GET url"

File: confluence/confluence-v1.json [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+                            "parentIdKey": "id"
+                        }
+                    }
+                ],
+                "rootRequest": {
+                    "dataRequest": {
+                        "endpoint": "/wiki/rest/api/content",
+                        "pagination": {
+                            "paginationByBatchSize": {
+                                "batchSize": 1,
```
Comment (mcondo) [inline confluence/confluence-v1.json:41]: Maybe it should be better to recommend a higher batchSize (5 or maybe the default 25?) , so less requests will be made. What do you think?

File: confluence/confluence-v1.asciidoc [hunk 9]
```diff
@@ -0,0 +1,73 @@
+Confluence Rest Configuration
+----------------------------
+This documentation describs what data the confluence configuration retrives and the endpoints used.
+This is this list of data that rest connector using confluence configuration will crawl.
+
+* spaces
```
Comment (mcondo) [inline confluence/confluence-v1.asciidoc:6]: In the datasource, I don't see the 'spaces' endpoint configured  `/wiki/rest/api/space` , does it need to be added now?

File: confluence/confluence-v1.json [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+                                {
+                                    "queryKey": "expand",
+                                    "queryValue": "body.storage"
+                                }
+                            ]
+                        },
+                        "childResponseMapping": {
+                            "childDataPath": "results",
+                            "parentIdKey": "id"
+                        }
```
Comment (mcondo) [inline confluence/confluence-v1.json:33]: children requests is also able to perform pagination, does it need to be configured now?

**Pull request merged**